.github/workflows/azure_deploy.yml

```diff
@@ -0,0 +1,151 @@
+name: Deploy to Zooniverse Static
+
+on:
+    # Run this workflow from other workflows
+    workflow_call:
+      inputs:
+        commit_id: 
+          description: 'HEAD commit hash'
+          required: true
+          type: string
+        deploy_url:
+          description: 'Deployed URL'
+          required: true
+          type: string
+        job_name:
+          description: 'Job name to report on'
+          required: true
+          type: string
+        script:
+          description: 'npm build script'
+          required: true
+          type: string
+        source:
+          default: './dist'
+          description: 'Source directory'
+          required: false
+          type: string
+        target:
+          description: 'Path to target folder'
+          required: true
+          type: string
+        title:
+          description: 'Notification title'
+          required: true
+          type: string
+      secrets:
+        creds:
+          required: true
+        slack_webhook_url:
+          required: true
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    env:
+      HEAD_COMMIT: ${{ inputs.commit_id }}
+    steps:
+    - uses: actions/checkout@v2
+    - uses: azure/login@v1
+      with:
+          creds: ${{ secrets.creds }}
+
+    - name: Cache dependencies
+      uses: actions/cache@v1
+      with:
+        path: ~/.npm
+        key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-node-
+
+    - name: Node.js build
+      id: build
+      uses: actions/setup-node@v1
+      with:
+        node-version: '14.x'
+    - run: npm ci
+    - run: npm run ${{ inputs.script }}
+
+    - name: Write commit_id.txt
+      run: echo ${HEAD_COMMIT} > ${{ inputs.source }}/commit_id.txt
+
+    - name: Upload to blob storage
+      id: upload
+      uses: azure/CLI@v1
+      with:
+        inlineScript: |
+            az storage blob upload \
+              --account-name zooniversestatic \
+              --content-cache-control 'public, max-age=60' \
+              --container-name '$web' \
+              --name '${{ inputs.target }}/index.html' \
+              --file '${{ inputs.source }}/index.html'
+            rm ${{ inputs.source }}/index.html
+            az storage blob upload \
+              --account-name zooniversestatic \
+              --content-cache-control 'public, max-age=60' \
+              --container-name '$web' \
+              --name '${{ inputs.target }}/commit_id.txt' \
+              --file '${{ inputs.source }}/commit_id.txt'
+            rm ${{ inputs.source }}/commit_id.txt
+            az storage blob upload-batch \
+              --account-name zooniversestatic \
+              --content-cache-control 'public, immutable, max-age=604800' \
+              --destination '$web/${{ inputs.target }}' \
+              --source ${{ inputs.source }}
+
+# Azure logout
```

Comment:
Logging out is only required on self hosted runners, per https://github.com/Azure/login#az-logout-and-security-hardening

Reply:
👍🏻